packages/select/src/Select.tsx

```diff
@@ -0,0 +1,488 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { IdAllocator } from '@leafygreen-ui/lib';
+import { breakpoints } from '@leafygreen-ui/tokens';
+import { colorSets, mobileSizeSet, Mode, Size, sizeSets } from './styleSets';
+import ListMenu from './ListMenu';
+import MenuButton from './MenuButton';
+import SelectContext from './SelectContext';
+import { InternalOption, OptionElement } from './Option';
+import {
+  convertToInternalElements,
+  getOptionValue,
+  isOptionDisabled,
+  isOptionSelectable,
+  reconcileOption,
+  traverseSelectChildren,
+  useStateRef,
+} from './utils';
+import { useViewportSize } from '@leafygreen-ui/hooks/src';
+
+const labelStyle = css`
+  display: inline-block;
+  margin-bottom: 2px;
+  font-weight: bold;
+`;
+
+export type Props = {
+  children: React.ReactNode;
+  className?: string;
+  id?: string;
+  darkMode?: boolean;
+  size?: Size;
+  disabled?: boolean;
+  label: string;
+  description?: string;
+  placeholder?: string;
+  name: string;
+} & (
+  | // Uncontrolled
+  ({
+      defaultValue?: string;
+      value?: undefined;
+    } & {
+      onChange?: (value: string) => void;
```

Comment:
mentioned above, but I think we should pass up the entire event for consistency-sake

Reply:
Still feel pretty strongly about this